### PR TITLE
DOC: clarification of default rvs method in scipy.stats

### LINKED ISSUE #9285: Mention sampling method for rvs in doc
It would be nice if the default method used for drawing samples (i.e. inverse transform sampling) would be mentioned somewhere in the documentation.

The source code explains it:
https://github.com/scipy/scipy/blob/v1.1.0/scipy/stats/_distn_infrastructure.py#L881
```
## Use basic inverse cdf algorithm for RV generation as default.
U = self._random_state.random_sample(self._size)
Y = self._ppf(U, *args)
return Y
```

But in my opinion this is a bit too hidden for something that is not at all obvious and more than just a trivial implementation detail.

Thanks :)

### PR DESCRIPTION
**Pull request by chrisb83**
Closes #9285

### CODE REVIEW

File: scipy/stats/_distn_infrastructure.py [hunk 1]
```diff
@@ -1407,7 +1407,16 @@ class rv_continuous(rv_generic):
...
+    The default method ``_rvs`` relies on the inverse of the cdf, ``_ppf``,
+    applied to a uniform random variate. In order to generate random variates
+    efficiently, either the default ``_ppf`` needs to be overwritten (e.g.
+    if the inverse cdf can expressed in an explicit form) or a sampling
+    method needs to be implemented in a custom ``_rvs`` method.
+
     Rarely would you override ``_isf``, ``_sf`` or ``_logsf``, but you could.
+    The main reason would be to improve numerical accuracy: for example,
+    the survival function ``_sf`` is computed as ``1 - _cdf`` which can
+    result in loss of precision if ``_cdf(x)`` is close to zero.
```
Comment (josef-pkt) [inline scipy/stats/_distn_infrastructure.py]: this should be: "is close to one"

File: scipy/stats/_distn_infrastructure.py [hunk 2]
```diff
@@ -1407,7 +1407,16 @@ class rv_continuous(rv_generic):
...
       _logpdf, _cdf, _logcdf, _ppf, _rvs, _isf, _sf, _logsf
 
+    The default method ``_rvs`` relies on the inverse of the cdf, ``_ppf``,
+    applied to a uniform random variate. In order to generate random variates
+    efficiently, either the default ``_ppf`` needs to be overwritten (e.g.
+    if the inverse cdf can expressed in an explicit form) or a sampling
+    method needs to be implemented in a custom ``_rvs`` method.
+
     Rarely would you override ``_isf``, ``_sf`` or ``_logsf``, but you could.
+    The main reason would be to improve numerical accuracy: for example,
```
Comment (WarrenWeckesser) [inline scipy/stats/_distn_infrastructure.py]: This looks fine as it is, but as long as you are here, how about dropping "rarely" from the previous sentence?  I don't think we should say "rarely", given that the default methods are guaranteed to lose precision in the tails.  Plus, of the 111 distributions (continuous and discrete) in scipy, we implement a custom `_sf` for 35 of them, so it is not "rare" in our own implementations.  (I'm also inclined to inspect the behavior of the other 76; they might benefit from implementing `_sf`.)

That sentence could be replaced by something that suggests it is normal, and in many cases even recommended, to override those methods, for the reason that you then explain.

**Comment by WarrenWeckesser** [comment]
This looks good.  I suggested a change in-line, but it doesn't have to be made in this PR.

**Comment by rgommers** [comment]
LGTM, thanks @chrisb83 @WarrenWeckesser

**Pull request merged**